go/database/mpt/io/live_test.go

```diff
@@ -287,3 +282,100 @@ func TestCheckEmptyDirectory_FailsIfDirectoryContainsADirectory(t *testing.T) {
 		t.Errorf("unexpected error: %v", err)
 	}
 }
+
+func TestIO_ExportBlockFromArchive(t *testing.T) {
+	// Create a small Archive from which we export LiveDB genesis.
+	sourceDir := t.TempDir()
+	archive, err := mpt.OpenArchiveTrie(sourceDir, mpt.S5ArchiveConfig, mpt.NodeCacheConfig{Capacity: 1024})
+	if err != nil {
+		t.Fatalf("failed to create archive: %v", err)
+	}
+	const (
+		exportedBlockHeight = uint64(2)
+		M                   = 10
+		N                   = 3
+	)
+
+	var expectedHashes []common.Hash
+
+	for i := 0; i < M; i++ {
+
+		code := []byte{1, 2, 3}
+		u := uint64(i)
+
+		update := common.Update{}
+		for j := 0; j < N; j++ {
+			newAddr := common.AddressFromNumber(i + j)
+
+			update.CreatedAccounts = append(update.CreatedAccounts, newAddr)
+			update.Balances = append(update.Balances, common.BalanceUpdate{Account: newAddr, Balance: amount.New(u)})
+			update.Nonces = append(update.Nonces, common.NonceUpdate{Account: newAddr, Nonce: common.ToNonce(u)})
+			update.Codes = append(update.Codes, common.CodeUpdate{Account: newAddr, Code: code})
+			update.Slots = append(update.Slots, common.SlotUpdate{Account: newAddr, Key: common.Key{1}, Value: common.Value{1}})
+		}
+
+		if i%2 != 0 {
```

Comment:
I somehow cannot fully understand this algorithm.  If I am not wrong the code above re-creates two addresses and creates a new one in each block:

```
block0:  addr0, adder, addr2 
blcok1: addr0, addr1, addr2, address

```
in `block1` - `addr1` and `addr2` were re-created (old keys removed, created new ones, but with the same values, i.e. no-one can really see the difference.  `addr3` is new.

Next you either update or delete address which matches current block. In this case `addr1` - if it gets updated, it is set with the same values, right? I.e. no one can really see the difference.  If it gets deleted, it has some merit - a real change is happening.  

Maybe I could not fully understand the algorithm, but all in all I think a much simpler version would do the job:

In each loop `i`,  keep creating still the same address `Address{j}`, but for account properties and key values set always `i` or `i+1`.  This way, you will have a new version of the same accounts in each block. 

You can have a condition that one of these three addresses is deleted in each block, not created. 

that it is

Reply:
Im fully aware what it does. I just wanted to make the test a bit random and I think it serves its purpose. Do we still want to simplify it...?

Reply:
If you confirm that I understand the algorithm right, then how is it "a bit random" if two third of changes actually do  a "change" in a way that the result is still the same? 

Yes, I think the algorithm I propose is much more straightforward and performs actually more changes than the original one. I prefer it.